File: apps/android/app/src/main/java/com/gepitech/soulconnection/data/Customer.kt
package com.gepitech.soulconnection.data

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

enum class Gender(val code: String) {
    MA("MA"),
    FE("FE"),
    OT("OT")
}

@Parcelize
data class Customer(
    val id: Int,
    val name: String?,
    val surname: String?,
    val email: String?,
    val phone: String?,
    val photo: String?,
    val birthdate: String?,
    val description: String?,
    val gender: Gender?,
    val address: String?,
    val country: String?
) : Parcelable {
    override fun hashCode(): Int {
        return (id.hashCode() +
                (name?.hashCode() ?: 0) +
                (surname?.hashCode() ?: 0) +
                (email?.hashCode() ?: 0) +
                (phone?.hashCode() ?: 0) +
                (photo?.hashCode() ?: 0) +
                (birthdate?.hashCode() ?: 0) +
                (description?.hashCode() ?: 0) +
                (gender?.hashCode() ?: 0) +
                (address?.hashCode() ?: 0) +
                (country?.hashCode() ?: 0))
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Customer

        if (id != other.id) return false
        if (name != other.name) return false
        if (surname != other.surname) return false
        if (email != other.email) return false
        if (phone != other.phone) return false
        if (photo != other.photo) return false
        if (birthdate != other.birthdate) return false
        if (description != other.description) return false
        if (gender != other.gender) return false
        if (address != other.address) return false
        if (country != other.country) return false

        return true
    }
}

File: apps/ios/SoulConnection/SoulConnection/Model/Customer.swift
//
//  Customer.swift
//  SoulConnection
//
//  Created by Matheo Coquet on 10/09/2024.
//

import Foundation

struct Customer: Codable, Identifiable {
    var id: Int
    var legacyId: Int?
    var email: String
    var name: String
    var surname: String
    var description: String
    var birthDate: String
    var gender: String
    var sign: String
    var phone: String?
    var photo: String?
    var photoFormat: String?
    var address: String?
    var coachId: Int?
    var createdAt: String
    var paymentMethods: [String]?
    var country: String?
}

struct CustomerPage: Codable {
    var index: Int
    var size: Int
    var totalPages: Int?
    var isLast: Bool
    var items: [Customer]
}
